### PR TITLE
Distinguishing more uses of CONST_RATIONAL

### PR DESCRIPTION
**Pull request by ajreynol**
Towards eliminating subtyping Int/Real.

### CODE REVIEW

File: src/proof/alethe/alethe_post_processor.cpp [hunk 1]
```diff
@@ -1874,10 +1874,7 @@ bool AletheProofPostprocessCallback::finalStep(
   if (id != PfRule::ALETHE_RULE)
   {
     std::vector<Node> sanitized_args{
-        res,
-        res,
-        nm->mkConst<Rational>(CONST_RATIONAL,
-                              static_cast<unsigned>(AletheRule::ASSUME))};
+        res, res, nm->mkConstInt(static_cast<unsigned>(AletheRule::ASSUME))};
```
Comment (nafur) [inline src/proof/alethe/alethe_post_processor.cpp:1877]:
```suggestion
        res, res, nm->mkConstInt(static_cast<uint32_t>(AletheRule::ASSUME))};
```

File: src/proof/alethe/alethe_post_processor.cpp [hunk 2]
```diff
@@ -1940,8 +1937,8 @@ bool AletheProofPostprocessCallback::addAletheStep(
   }
 
   std::vector<Node> new_args = std::vector<Node>();
-  new_args.push_back(NodeManager::currentNM()->mkConst(
-      CONST_RATIONAL, Rational(static_cast<unsigned>(rule))));
+  new_args.push_back(NodeManager::currentNM()->mkConstInt(
+      Rational(static_cast<unsigned>(rule))));
```
Comment (nafur) [inline src/proof/alethe/alethe_post_processor.cpp:1941]:
```suggestion
      Rational(static_cast<uint32_t>(rule))));
```

File: src/theory/arith/nl/ext/factoring_check.cpp [hunk 3]
```diff
@@ -35,7 +35,6 @@ namespace nl {
 FactoringCheck::FactoringCheck(Env& env, ExtState* data)
     : EnvObj(env), d_data(data)
 {
-  d_zero = NodeManager::currentNM()->mkConst(CONST_RATIONAL, Rational(0));
   d_one = NodeManager::currentNM()->mkConst(CONST_RATIONAL, Rational(1));
```
Comment (nafur) [inline src/theory/arith/nl/ext/factoring_check.cpp:38]: What about that one?

**Review by nafur — APPROVED** [review]
LGTM.

File: src/theory/arith/nl/ext/factoring_check.cpp (hunk 3 above)
Comment (ajreynol) [reply src/theory/arith/nl/ext/factoring_check.cpp:38]: This is harder to address, I'll deal with this in another PR.

**Pull request merged**